app/src/main/java/ch/epfl/sdp/artificial_intelligence/Enemy.java

```diff
@@ -2,32 +2,51 @@
 
 import java.util.List;
 
-import ch.epfl.sdp.Player;
+import ch.epfl.sdp.entity.EntityType;
+import ch.epfl.sdp.entity.Player;
+import ch.epfl.sdp.entity.PlayerManager;
 
-public class Enemy extends MovingEntity {
+public class Enemy extends MovingArtificialEntity {
     private Behaviour behaviour;
     private List<Player> players; // For now I use a list of players, but it could be nice to have
     // a static manager of players.
     private int damage;
     private float dps; // damage per second
-    private float detectionDistance;
+    private double detectionDistance;
     private int timeAttack;
     private int timeWandering;
     private LocalBounds patrolBounds;
-    private Boundable maxBounds;
     private boolean waiting;
 
-    public Enemy(List<Player> players, int damage, float dps, float detectionDistance, LocalBounds patrolBounds, Boundable maxBounds) {
+    public Enemy() {
+        super(new RectangleBounds(50, 50, null));
+        super.setAoeRadius(1);
+        super.setVelocity(1);
+        super.setMoving(true);
+        this.damage = 1;
+        this.dps = 1;
+        this.detectionDistance = 1;
+        this.players = PlayerManager.getPlayers();
+        behaviour = Behaviour.PATROL;
+        timeAttack = 100; // Needs calibration
+        timeWandering = 100;
+        this.patrolBounds = new LocalBounds(new UnboundedArea(), getPosition());
+        this.waiting = false;
+    }
+
+    public Enemy(int damage, float dps, float detectionDistance, double aoeRadius, LocalBounds patrolBounds, Boundable maxBounds) {
         this.damage = damage;
         this.dps = dps;
         this.detectionDistance = detectionDistance;
-        this.players = players;
+        this.players = PlayerManager.getPlayers();
         behaviour = Behaviour.WAIT;
         timeAttack = 100; // Needs calibration
         timeWandering = 100;
         this.patrolBounds = patrolBounds;
-        this.maxBounds = maxBounds;
         this.waiting = false;
+        if (aoeRadius < detectionDistance) {
+            this.setAoeRadius(aoeRadius);
```

Comment:
What if the aoeRadius >= detectionDistance, how will the aoeRadius be initialized?